commands/audit.go

```diff
@@ -0,0 +1,242 @@
+	}
+
+	// Audit user policy file if it exists and not skipping
+	if !a.SkipUserPolicy {
+		userPolicyPath, err := a.getUserPolicyPath()
+		if err == nil && userPolicyPath != "" {
+			userResults, userExists, err := a.auditPolicyFileWithStatus(userPolicyPath, validator)
+			if err != nil {
+				fmt.Fprintf(a.Out, "WARNING: Failed to audit user policy file at %s: %v\n", userPolicyPath, err)
+				// Don't fail completely if user policy is unreadable
+			} else if userExists {
+				fmt.Fprintf(a.Out, "\nValidating %s...\n\n", userPolicyPath)
+				for _, result := range userResults {
+					a.printResult(result)
+				}
+				allResults = append(allResults, userResults...)
+			}
+		}
+	}
+
+	// Print summary only (results already printed above)
+	if len(allResults) == 0 {
+		fmt.Fprintf(a.Out, "\nNo policy entries to validate.\n")
+	}
+
+	// Print summary
+	summary := policy.CalculateSummary(allResults)
+	a.printSummary(summary)
+
+	// Return appropriate exit code
+	return summary.GetExitCode()
+}
+
+// auditPolicyFileWithStatus validates all entries in a policy file and returns results, whether file exists, and any errors
+func (a *AuditCmd) auditPolicyFileWithStatus(policyPath string, validator *policy.PolicyValidator) ([]policy.ValidationResult, bool, error) {
+	var results []policy.ValidationResult
+
+	// Check if file exists
+	exists, err := afero.Exists(a.Fs, policyPath)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to check if policy file exists: %w", err)
+	}
+
+	if !exists {
+		// File doesn't exist, return empty results with exists=false
+		return results, false, nil
+	}
+
+	// Load policy file
+	content, err := afero.ReadFile(a.Fs, policyPath)
+	if err != nil {
+		return nil, true, fmt.Errorf("failed to read policy file: %w", err)
+	}
+
+	// Parse policy
+	p := policy.FromTable(content, policyPath)
+
+	lineNumber := 1
+	for _, user := range p.Users {
+		// Each user entry maps to principals
+		for _, principal := range user.Principals {
+			result := validator.ValidateEntry(principal, user.IdentityAttribute, user.Issuer, lineNumber)
+			results = append(results, result)
+		}
+		lineNumber++
+	}
+
+	return results, true, nil
+}
+
+
+// getUserPolicyPath returns the path to the user's policy file, or empty string if not found
+func (a *AuditCmd) getUserPolicyPath() (string, error) {
+	if a.CurrentUsername == "" {
+		return "", nil
+	}
+
+	u, err := a.UserPolicyLookup.Lookup(a.CurrentUsername)
+	if err != nil {
+		// User not found, return empty (not an error)
+		return "", nil
+	}
+
+	userPolicyPath := filepath.Join(u.HomeDir, ".opk", "auth_id")
+	return userPolicyPath, nil
+}
+
+// printResult prints a single validation result
+func (a *AuditCmd) printResult(result policy.ValidationResult) {
+	var statusSymbol string
+	switch result.Status {
+	case policy.StatusSuccess:
+		statusSymbol = "✓"
+	case policy.StatusWarning:
+		statusSymbol = "⚠"
+	case policy.StatusError:
+		statusSymbol = "✗"
```

Comment:
It is a reasonable choice to use unicode characters here. They look read well and are informative. I try not to use any non-ascii characters in opkssh output because of concerns over terminals and shells that many not support unicode well.